### DISCUSSION TITLE
Redirect with params?

### DISCUSSION THREAD

**Discussion by juankamilo** [nitrojs/nitro#1280] [Q&A]
Can nitro handle redirect with params?
example:
http://localhost:8080/es/test/oldroute?testparam=123
to
http://localhost:8080/es/new/route?testparam=123

I don't see the option in code/docus so I guess not, but it would be nice to have the option.

**Comment by maou-shonen** [comment]
h3
```ts
sendRedirect(event, location, code=302)
```
You should be able to use any url string into the location

**Reply by juankamilo** [reply]
Hi maou
Thx for the help, the idea was to put it in the routeRules to prevent the page load (it is posible?), I handled with a navigateTo but is not the optimal.
This you suggest would be to handle it through a middleware I assume?

**Reply by maou-shonen** [reply]
Certainly, you can achieve that using routeRules,
or you can use sendRedirect in the routes section to accomplish more complex tasks.
